main.py

```diff
@@ -0,0 +1,654 @@
+    
+    #Default Constructor
+    def __init__(self):
+        self.cellsInParallel = 0
+        self.cellsInSeries = 0
+        self.energyRequired = 0
+        self.voltageRequired = 0
+        self.currentRequired = 0
+        self.peakCurrentRequired= 0
+        self.additionalCapacity = 30
+        self.currentCell = cell.emptyCell()
+
+    #Set energy spec
+    def setEnergyRequired(self, energy):
+        self.energyRequired = energy
+
+    #Sets voltage spec
+    def setVoltageRequired(self, myVoltage):
+        self.voltageRequired = myVoltage
+
+    #Sets power spec
+    def setPowerRequired(self,powerInKW):
+        self.powerRequired = powerInKW
+    
+    #Sets capacity margin
+    def setAdditionalCapacity(self, percentage):
+        self.additionalCapacity = percentage
+
+    #Sets total cell count
+    def setTotalCells(self, cellCount):
+        self.totalCells = cellCount
+
+    #Sets weight battery pack weight
+    def setWeightInKilograms(self, weight):
+        self.weightInKilograms = weight
+
+    #Sets cell 
+    def setCell(self, newCell):
+        self.currentCell = newCell
+
+    def addCellToList(self, newCell):
+        self.cellList.append(newCell)
+
+    def getEnergyRequired(self):
+        if(FLAGS_ENABLED ==  1):
+            if(self.energyRequired <= 0):
+                print('Warning: Function getEnergyRequired() -- member of class pack -- enerrgyRequried should be greater than 0')
+        return self.energyRequired
+
+    def getVoltageRequired(self):
+        if(FLAGS_ENABLED == 1):
+            if(self.voltageRequired <= 0):
+                print('Warning: Function getVoltageRequired() -- member of class pack -- voltageRequired should be greater than 0')
+        return self.voltageRequired
+    
+    def getPowerRequired(self):
+        if(FLAGS_ENABLED == 1):
+            if(self.getPowerRequired <= 0):
+                print('Warning: Function getPowerRequired() -- member of class pack -- power requried should be greater than 0')
+        return self.powerRequired
+
+    def getAdditionalCapacity(self):
+        return self.additionalCapacity
+
+    def getTotalCells (self):
+        if(FLAGS_ENABLED ==1):
+            if(self.totalCells <= 0):
+                print('Function: getTotalCells in pack -  Warning: cell count is incorrect')
+        return self.totalCells
+
+    def getWeight(self):
+        if(FLAGS_ENABLED == 1):
+            if(self.weightInKilograms <= 0):
+                print('Warning: -- Function getWeight() -- member of class pack -- weightInKilograms is less than or equal to 0')
+
+        return  self.weightInKilograms
+
+    def getCell(self):
+        return self.currentCell
+
+    def getCellsInSeries(self):
+        return self.cellsInSeries
+
+    def getCellsInParallel(self):
+
+        return self.cellsInParallel
+
+    #Gets capacity in Ah
+    def getCapacity(self):
```

Comment:
Just an exception handling reminder - if the value is not possible, after printing set the value to a null value like -1 so the user knows it could not be calculated